### ISSUE TITLE
Conan imports does not work as expected for local packages

### ISSUE THREAD

**Issue by AliAskari** [conan-io/conan#1642]
As [per document](http://conanio.readthedocs.io/en/latest/reference/commands/imports.html#conan-imports) calling conan imports on a package that is inside local cache should call the `imports` method of the recipe i.e. Calling `conan imports pkgA/1.0.0@user/channel` where `pkgA/1.0.0@user/channel` is in the local cache should call the `imports` method declared in conanfile.py.

However, when I call imports on a such package I get the following error:

    ERROR: Execute 'conan install -g txt' first.

and obviously calling `conan install -g txt` in folder without a conan file will fail. Am I not understanding the documentation properly or the imports doesn't work as expected yet?

This is the commands that I execute on my machine:

    conan install pkgA/1.0.0@user/channel => Downloads the package to my local cache
    conan search pkgA/*                   => Returns with the package in my local cache, so indeed the package is downloaded and it is in my cache   
    conan imports pkgA/1.0.0@user/channel => fails with the error message mentioned above

**Comment by lasote** [comment]
I agree there is something weird here. Let me take a look.
What is your use case to call the imports of a recipe in the local cache? Where do you expect to receive the files? In a local directory?

**Comment by lasote** [comment]
It is a bug, I'll fix it and include it in 0.26. But please, tell me your use case to understand better.

**Comment by lasote** [comment]
It's automatically fixed in the PR: https://github.com/conan-io/conan/pull/1625
Still pending and opened to a discussion but assigned to 0.26.

You will need to run first:

`conan install pkgA/1.0.0@user/channel -g txt`

**Comment by AliAskari** [comment]
I see, thanks for the quick fix.

The main reason I tried to use it was that my build machine is not the same as the machine that I use to run my application. So, even though I install the package on my build machine, using `conan install`, the shared libraries are useless here. What I actually want to happen is to have the shared libraries on a different machine. The copy part is done using a post build script at the moment but I think `conan imports` is a more sophisticated method to handle shared libraries because I don't have to worry about anything.

Another reason is my CI. In my CI I run the application in a different docker container than the one I built the application. I cannot cache the `/usr/lib` (that's where import the shared libs to) folder between different stages in my CI. 

In both cases I can still call conan install, but it installs **all** dependencies which is not efficient. If `conan imports` work I can just install the few runtime packages that I need and then call `imports` and I am done. It saves around 5 minutes in my CI pipeline.
